File: src/week2/kim_sung_hyun/가장 큰 수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        class cmp implements Comparator<String>{
            public int compare(String a, String b){
                return (b+a).compareTo(a+b);
            }
        }
        String answer = "";
        String [] nTostr = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            nTostr[i] = Integer.toString(numbers[i]);
        }
        Arrays.sort(nTostr,new cmp());
        if(nTostr[0].equals("0"))
            return "0";
        for (int i = 0; i < nTostr.length; i++) {
            answer+=nTostr[i];
        }

        return String.join("",nTostr);
    }
}

File: src/week2/kim_hyun_jun/가장_큰_수.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

bool compare(string a, string b) {
	return a + b > b + a;
}

string solution(vector<int> numbers) {
	string answer = "";
	vector<string> v;
	for (auto e : numbers) {
		v.push_back(to_string(e));
	}

	sort(v.begin(), v.end(), compare);

	if (v[0] == "0") return "0";

	for (auto e : v) {
		answer += e;
	}

	return answer;
}